File: backend/src/main/java/com/back/ourlog/domain/user/dto/MyProfileResponse.java
package com.back.ourlog.domain.user.dto;

import com.back.ourlog.domain.user.entity.User;

public record MyProfileResponse(
        Integer userId,
        String email,
        String nickname,
        String profileImageUrl,
        String bio,
        Integer followingsCount,
        Integer followersCount
) {
    public static MyProfileResponse from(User user) {
        return new MyProfileResponse(
                user.getId(),
                user.getEmail(),
                user.getNickname(),
                user.getProfileImageUrl(),
                user.getBio(),
                user.getFollowingsCount(),
                user.getFollowersCount()
        );
    }
}

File: backend/src/main/java/com/back/ourlog/domain/user/dto/MyProfileResponse.kt
package com.back.ourlog.domain.user.dto

import com.back.ourlog.domain.user.entity.User

data class MyProfileResponse(
    val userId: Int?,
    val email: String?,
    val nickname: String?,
    val profileImageUrl: String?,
    val bio: String?,
    val followingsCount: Int?,
    val followersCount: Int?
) {
    companion object {
        fun from(user: User): MyProfileResponse {
            return MyProfileResponse(
                user.id,
                user.email,
                user.nickname,
                user.profileImageUrl,
                user.bio,
                user.followingsCount,
                user.followersCount
            )
        }
    }
}